server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1474,7 +1474,15 @@ private DateMathExpressionResolver() {
 
         public static List<String> resolve(final Context context, List<String> expressions) {
             List<String> result = new ArrayList<>(expressions.size());
-            for (String expression : expressions) {
+            boolean wildcardSeen = false;
+            for (int i = 0; i < expressions.size(); i++) {
+                String expression = expressions.get(i);
```

Comment:
I'm assuming it's impossible to get an empty expression here? e.g "".

Reply:
I've refactored it such that `expressions` can be empty.